massa-grpc/src/tests/stream.rs

```diff
@@ -1217,8 +1181,162 @@ async fn new_slot_execution_outputs() {
+                grpc_api::ExecutionEventFilter {
+                    filter: Some(grpc_api::execution_event_filter::Filter::None(
+                        grpc_model::Empty {},
+                    )),
+                },
+            ),
+        ),
+    };
+    filters.push(filter);
+    // async pool changes filter
+    let can_be_executed = match async_pool_changes.0.iter().next().unwrap().1 {
+        massa_ledger_exports::SetUpdateOrDelete::Set(value) => Some(value.can_be_executed),
+        massa_ledger_exports::SetUpdateOrDelete::Update(value) => match value.can_be_executed {
+            massa_ledger_exports::SetOrKeep::Set(value) => Some(value),
+            massa_ledger_exports::SetOrKeep::Keep => None,
+        },
+        massa_ledger_exports::SetUpdateOrDelete::Delete => None,
+    };
+
+    filter = grpc_api::NewSlotExecutionOutputsFilter {
+        filter: Some(
+            grpc_api::new_slot_execution_outputs_filter::Filter::AsyncPoolChangesFilter(
+                grpc_api::AsyncPoolChangesFilter {
+                    filter: Some(grpc_api::async_pool_changes_filter::Filter::CanBeExecuted(
+                        can_be_executed.unwrap(),
+                    )),
+                },
+            ),
+        ),
+    };
+    filters.push(filter);
+
+    // executed denunciations changes filter
+    filter = grpc_api::NewSlotExecutionOutputsFilter {
+        filter: Some(
+            grpc_api::new_slot_execution_outputs_filter::Filter::ExecutedDenounciationFilter(
+                grpc_api::ExecutedDenounciationFilter {
+                    filter: Some(grpc_api::executed_denounciation_filter::Filter::None(
+                        grpc_model::Empty {},
+                    )),
+                },
+            ),
+        ),
+    };
+    filters.push(filter);
+    let (op_id, (_, _)) = executed_ops_changes.iter().next().unwrap();
+    // executed operations changes filter
+    filter = grpc_api::NewSlotExecutionOutputsFilter {
+        filter: Some(
+            grpc_api::new_slot_execution_outputs_filter::Filter::ExecutedOpsChangesFilter(
+                grpc_api::ExecutedOpsChangesFilter {
+                    filter: Some(grpc_api::executed_ops_changes_filter::Filter::OperationId(
+                        op_id.to_string(),
+                    )),
+                },
+            ),
+        ),
+    };
+    filters.push(filter);
+
+    // ledger changes filter
+    filter = grpc_api::NewSlotExecutionOutputsFilter {
+        filter: Some(
+            grpc_api::new_slot_execution_outputs_filter::Filter::LedgerChangesFilter(
+                grpc_api::LedgerChangesFilter {
+                    filter: Some(grpc_api::ledger_changes_filter::Filter::None(
+                        grpc_model::Empty {},
+                    )),
+                },
+            ),
+        ),
+    };
+    filters.push(filter);
+
+    tx_request
+        .send(NewSlotExecutionOutputsRequest { filters })
```

Comment:
to improve readability please remove the vector `filters` and create the vec here